pymeasure/instruments/voltcraft/voltcraft_lcr500.py

```diff
@@ -0,0 +1,162 @@
+    def __init__(self, adapter, name="Voltcraft LCR-500", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+        self.adapter.connection.write_termination = "\n"
+        self.adapter.connection.read_termination = "\n"
+
+    def _parse_fetch(self, value):
+        """Parse the fetch response into a tuple of floats.
+        The response is expected to be in the format: 'primary,secondary,range'
+        """
+        parts = value.split(",")
+        if len(parts) != 3:
+            raise ValueError("Invalid fetch response: %s" % value)
+        return float(parts[0]), float(parts[1]), int(parts[2])
+
+    go_to_local = Instrument.setting(
+        "*GTL", """Switch the device to local mode and disable key lock."""
+    )
+
+    # -- Frequency Subsystem --
+    frequency = Instrument.control(
+        "FREQ?",
+        "FREQ %d",
+        """Control the measurement frequency in Hz (int).
+        Can be one of: 100, 120, 400, 1k, 4k, 10k, 40k, 50k, 75k, 100k.
+        """,
+        validator=strict_discrete_set,
+        values=_FREQS_HZ,
+    )
+
+    # -- Function Subsystem --
+    main_parameter = Instrument.control(
+        "FUNC:IMPA?",
+        "FUNC:IMPA %s",
+        """Control the main measurement parameter (str).
+        Can be one of: 
+            'R': Resistance, 
+            'L': Inductance, 
+            'C': Capacitance, 
+            'Z': Impedance, 
+            'AUTO': Automatic.
+        """,
+        validator=strict_discrete_set,
+        values=_IMPA,
+    )
+
+    secondary_parameter = Instrument.control(
+        "FUNC:IMPB?",
+        "FUNC:IMPB %s",
+        """Control the secondary measurement parameter (str).
+        Can be one of:
+            'X': Reactance, 
+            'Q': Quality factor, 
+            'D': Dissipation factor, 
+            'THETA': Phase angle, 
+            'ESR': Equivalent series resistance.
+        """,
+        validator=strict_discrete_set,
+        values=_IMPB,
+    )
+
+    measurement_range = Instrument.control(
+        "FUNC:RANG?",
+        "FUNC:RANG %s",
+        """Control the measurement range (str or int).
+        Can be one of: 'AUTO', 10, 100, 1k, 10k, 100k.
+        """,
+        validator=mixed_validator,
+        values=_RESIS_OHM,
+    )
+
+    level = Instrument.control(
+        "FUNC:LEV?",
+        "FUNC:LEV %d",
+        """Control the measurement signal level in mVrms (int).
+        Can be one of: 300, 600.
+        """,
+        validator=strict_discrete_set,
+        values=_LEVELS_MVRMS,
+    )
+    equivalent_circuit = Instrument.control(
+        "FUNC:EQUI?",
+        "FUNC:EQUI %s",
+        """Control the equivalent circuit type (str).
+        Can be one of: 'SER', 'PAL'.
+        """,
+        validator=strict_discrete_set,
+        values=_EQUIV,
+    )
+
+    # -- Fetch Subsystem --
+    fetch = Instrument.measurement(
+        "FETC?",
+        """Perform a measurement and return the result as tuple of
```

Comment:
Docstrings should start with Get/Measure for a `measurement` and Set for `setting` and Control for `Control`.